Fix flaky test

The `TestCreateStreamNoMetadataLeader` unit test fails occasionally and should be fixed.

```
panic: failed to checkpoint high watermark: cannot create temp file: open /tmp/liftbridge_test_103530399/b/streams/foo/0/replication-offset-checkpoint862222957: no such file or directory
```

I believe this particular error is due to removing `storagePath`.

Though the actual error I was referring to is this one [here](https://circleci.com/gh/liftbridge-io/liftbridge/1007).

```
--- FAIL: TestCreateStreamNoMetadataLeader (7.49s)
    api_test.go:129: 
        	Error Trace:	api_test.go:129
        	Error:      	Not equal: 
        	            	expected: "no known metadata leader"
        	            	actual  : "context deadline exceeded"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-no known metadata leader
        	            	+context deadline exceeded
        	Test:       	TestCreateStreamNoMetadataLeader
```

I think I know what the issue is. Will take a look.

--- a/server/api_test.go
+++ b/server/api_test.go
@@ -116,7 +116,7 @@ func TestCreateStreamNoMetadataLeader(t *testing.T) {
 	// a server and wait for the leader to step down.
 	getMetadataLeader(t, 10*time.Second, s1, s2)
 	s1.Stop()
-	waitForNoMetadataLeader(t, 10*time.Second, s1, s2)
+	waitForNoMetadataLeader(t, 10*time.Second, s2)
 
 	// Connect and send the request to the follower.
 	client, err := lift.Connect([]string{"localhost:5050"})

--- a/server/server_test.go
+++ b/server/server_test.go
@@ -84,6 +84,17 @@ func getMetadataLeader(t *testing.T, timeout time.Duration, servers ...*Server) 
 	if leader == nil {
 		stackFatalf(t, "No metadata leader found")
 	}
+
+	// Wait for cluster to agree on leader.
+LOOP:
+	for time.Now().Before(deadline) {
+		for _, s := range servers {
+			if string(s.getRaft().Leader()) != leader.config.Clustering.ServerID {
+				continue LOOP
+			}
+		}
+		break
+	}
 	return leader
 }
 

